src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -326,48 +678,43 @@ private bool WriteMemberText(out object o, Member anyText)
                     {
                         if (text.Mapping.TypeDesc.CollapseWhitespace)
                         {
-                            o = CollapseWhitespace(ReadString());
+                            value = CollapseWhitespace(ReadString());
                         }
                         else
                         {
-                            o = ReadString();
+                            value = ReadString();
                         }
                     }
                     else
                     {
                         if (text.Mapping.TypeDesc == StringTypeDesc || text.Mapping.TypeDesc.FormatterName == "String")
                         {
-                            o = ReadString(null, text.Mapping.TypeDesc.CollapseWhitespace);
+                            value = ReadString(null, text.Mapping.TypeDesc.CollapseWhitespace);
                         }
                         else
                         {
-                            o = WritePrimitive(text.Mapping, () => ReadString());
+                            value = WritePrimitive(text.Mapping, () => ReadString());
                         }
                     }
                 }
 
+                anyText.Source.Invoke(value);
```

Comment:
>Invoke(value); [](start = 31, length = 14)

If Source is a MethodInfo, then we should probably create a delegate on Member to avoid the Invoke overhead. e.g. Member.SourceDelegate. Not necessary for this PR.